adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java

```diff
@@ -437,8 +437,10 @@ private void updatePaymentRequestForAlternateMethod(PaymentsRequest paymentsRequ
         paymentsRequest.setReturnUrl(cartData.getAdyenReturnUrl());
         if (ISSUER_PAYMENT_METHODS.contains(adyenPaymentMethod)) {
             paymentMethod.setIssuer(cartData.getAdyenIssuerId());
-        } else if (adyenPaymentMethod.startsWith(KLARNA)||adyenPaymentMethod.startsWith(PAYMENT_METHOD_FACILPAY_PREFIX)) {
-            setOpenInvoiceData(paymentsRequest, cartData, customerModel);
+        } else if (adyenPaymentMethod.startsWith(KLARNA)
+                || adyenPaymentMethod.startsWith(PAYMENT_METHOD_FACILPAY_PREFIX)
+                || OPENINVOICE_METHODS_API.contains(adyenPaymentMethod)) {
```

Comment:
I am not sure, but I think the OPENINVOICE_METHODS_API list was used in the past for sending payment requests to the old PAL /authorise endpoint for a few alternative payment methods(ratepay/klarna), if that's the case, I don't think we should use that list.

Reply:
The main reason I did this change is because Affirm requires `shopperName` to be passed on the PaymentsRequest. The only method where we set this value is `setOpenInvoiceData`, so I thought it made sense to use since this is indeed an OpenInvoice method.

It's also not changing the behaviour of any of the other payment methods:
- Klarna is already part of the list, like @rkewlani mentioned, I was just not sure why it is checked with `.startsWith(KLARNA)`. If there's no case other than "klarna" and "klarna_account" that will fall into that, we can remove and use only the OPENINVOICE_METHODS_API list.
- Ratepay, which is the other method on the list, is following a totally different flow that will use /authorise like Alessio mentioned. It's the only payment method using that right now, and I'm also not sure why. But anyway, for now it won't follow this flow.